src/collections/Message/Message.js

```diff
@@ -1,66 +1,198 @@
-import React, { Component, PropTypes } from 'react'
-import classNames from 'classnames'
-import $ from 'jquery'
-import META from '../../utils/Meta'
-import { getUnhandledProps } from '../../utils/propUtils'
-import Header from '../../elements/Header/Header'
+import _ from 'lodash'
+import React, { PropTypes } from 'react'
+import cx from 'classnames'
+
 import Icon from '../../elements/Icon/Icon'
+import MessageContent from './MessageContent'
+import MessageHeader from './MessageHeader'
+import MessageList from './MessageList'
+import MessageItem from './MessageItem'
 
-export default class Message extends Component {
-  static propTypes = {
-    children: PropTypes.node,
-    className: PropTypes.string,
-    dismissable: PropTypes.bool,
-    header: PropTypes.string,
-    icon: PropTypes.string,
-  }
+import META from '../../utils/Meta'
+import {
+  customPropTypes, getUnhandledProps, iconPropRenderer, useKeyOnly, useKeyOrValueAndKey,
+} from '../../utils/propUtils'
+import * as sui from '../../utils/semanticUtils'
 
-  componentDidMount() {
-    this.messageElm = $(this.refs.message)
-  }
+/**
+ * A message displays information that explains nearby content
+ */
+function Message(props) {
+  const {
+    children,
+    className,
+    content,
+    header,
+    icon,
+    list,
+    dismissable,
+    onDismiss,
+    hidden,
+    visible,
+    floating,
+    compact,
+    attached,
+    warning,
+    info,
+    positive,
+    success,
+    negative,
+    error,
+    color,
+    size,
+  } = props
 
-  handleDismiss = e => {
-    this.messageElm.transition('fade')
-  }
+  const classes = cx(
+    'ui',
+    size,
+    color,
+    useKeyOnly(icon, 'icon'),
+    useKeyOnly(hidden, 'hidden'),
+    useKeyOnly(visible, 'visible'),
+    useKeyOnly(floating, 'floating'),
+    useKeyOnly(compact, 'compact'),
+    useKeyOrValueAndKey(attached, 'attached'),
+    useKeyOnly(warning, 'warning'),
+    useKeyOnly(info, 'info'),
+    useKeyOnly(positive, 'positive'),
+    useKeyOnly(success, 'success'),
+    useKeyOnly(negative, 'negative'),
+    useKeyOnly(error, 'error'),
+    'message',
+    className,
+  )
 
-  static _meta = {
-    name: 'Message',
-    type: META.type.collection,
-  }
+  const dismissIcon = (dismissable || onDismiss) && <Icon name='close' onClick={onDismiss || _.noop} />
```

Comment:
FWIW I toyed with the idea of allowing auto controlled state here to toggle the `hide` prop.  In this case, it would make sense to just say "dismissable" and not handle clicks yourself. However, it seemed like a bad practice to let users hide Messages but leave them in memory.  I think it is better to manage a message stack yourself somewhere.